src/App.jsx

```diff
@@ -1,31 +1,27 @@
-import { Route, Routes } from "react-router-dom";
+import { Route, Routes, useLocation } from "react-router-dom";
 import { ToastContainer, Zoom } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
 import "./App.css";
-import {
-  HomePage,
-  ProductListing,
-  Cart,
-  Wishlist,
-  Login,
-  SignUp,
-} from "./pages";
+import { Header } from "./components";
+import { HomePage, ProductListing, Cart, Wishlist, Login, SignUp } from "./pages";
 
 function App() {
+  const { pathname } = useLocation();
   return (
     <div className="App">
       <ToastContainer
```

Comment:
you can create diffferent file for toast portal, also can create custom-hook from this, way cleaner!

Reply:
react-toastify lib handles the toasts. So, there's no need for me to have portals & hooks for the same.